Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -779,6 +779,66 @@ public struct Timestamp: BSONValue, Equatable, Codable {
     }
 }
 
+enum BSONEqualsError: Error { case InvalidArrayArgument(String) }
```

Comment:
The question of whether to throw an error for arrays or trigger a `preconditionFailure` is something I had to wrestle with. I settled on throwing an error, because it seemed excessive to halt the user's application if they called this function unaware of the fact that, even though `Array` does conform to `BSONValue`, our function accepting two `BSONValue`'s does not accept all kinds of Arrays. This is not really the same as feeding in a `bsonType` that should never really arise, such as `.undefined`, so an error seemed much more tame.

The downside is that we add a lone, random error case that is used and thrown only once in the entire codebase, and, now the user must wrap their calls to `bsonEquals()` with a `try`.

Feedback is welcome. :)